File: tree/tree.java
package tree;
import java.util.*;

public class tree {
    
    static class Node{
        int data;
        Node left,right;
        Node(int item){
            data = item;
            left = right = null;

        }
    }

    class BinaryTree{
        Node root;
        
    }
}
Stack<Integer> s = new Stack<Integer>();
Scanner sc = new Scanner(System.in);
int n = sc.nextInt();

int[] arr = new int[n];

File: DSAC/tree.cpp
#include<bits/stdc++.h>
using namespace std;


struct Node{
    int data;
    Node* left;
    Node* right;
    Node(int val) : data(val), left(nullptr),right(nullptr) {}
};

void preorder(Node* root){
    if(root == nullptr) return ;
    cout << root->data << " "; 
    preorder(root->left);
     
    preorder(root->right);
}

int main(){

    Node* root = new Node(1);
    root->left = new Node(2);
    root->right = new Node(3);
    root->left->left = new Node(4);
    root->left->right = new Node(0);
    cout << "Binary tree uccessfully." << endl;
    preorder(root);
    cout << endl;



}